exasol_advanced_analytics_framework/lua/src/event_loop_main.lua

```diff
@@ -0,0 +1,63 @@
+---
+-- @module event_loop_main
+--
+-- This scrip includes the main function of the Event Loop
+--
+
+local M = {}
+local exaerror = require("exaerror")
+local event_loop = require("event_loop")
+local json = require('cjson')
+
+_G.global_env = {
+    error = error
+}
+
+
+---
+-- Parse a given arguments in json string format.
+--
+-- @param json_str	input parameters as json string
+--
+-- @return lua table including parameters
+--
+function M._parse_arguments(json_str)
+    local success, args =  pcall(json.decode, json_str)
+    if not success then
+		local error_obj = exaerror.create(
+                "E-SME-5",
+                "It could not be converted to json object"
+        ):add_mitigations("Check syntax of the input string json is correct")
+		_G.global_env.error(tostring(error_obj))
+	end
+    return args
+end
+
+---
+-- Prepare the initial query that initiates the Event Loop and calls Event Handler
+--
+function M._prepare_init_query(args)
+    local action = args['action'] -- TODO
+    local params = args['parameters']
+    local schema = args['schema']
+    local sql_prefix = args['sql_prefix']
+
+    local _udf_name = schema .. "." .. sql_prefix .."_EVENT_HANDLER_RUNNER_UDF"
```

Comment:
Maybe it is enough, when we specify which UDF should be called. This also means, the user can define its own UDFs which can be used with this event loop. Perhaps, also in different languages. As long, the UDF follows the behavioral contract, it doesn't matter. We can have of course have one predefined UDF which awaits an action (Python Class) in its parameter.

With this we only need the following arguments

parameters
schema
udf_name
bucketfs_connection